### PR TITLE
Release-1.4.0

### PR DESCRIPTION
**Pull request by slim-bean**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,13 +1,227 @@
-## master / unreleased
+## 1.4.0 (2020-03-12)
 
-### Features
+Over 130 PR's merged for this release, from 40 different contributors!!  We continue to be humbled and thankful for the growing community of contributors and users of Loki.  Thank you all so much.
+
+### Important Notes
+
+Before we get into new features, version 1.4.0 brings with it the first (that we are aware of) upgrade dependency.
```
Comment (rfratto) [inline CHANGELOG.md:7]: Can we bring more attention to this somehow?

File: docs/operations/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+```yaml
+kvstore:
+  # The backend storage to use for the ring. Supported values are
+  # consul, etcd, inmemory
+  store: <string>
+```
+
+* Running without using a shared ring (inmemory): No action required
+* Running with a shared ring and upgrading from v1.3.0 -> v1.4.0: No action required
+* Running with a shared ring and upgrading from any version less than v1.3.0 (e.g. v1.2.0) -> v1.4.0: ACTION REQUIRED
```
Comment (rfratto) [inline docs/operations/upgrade.md:28]:
```suggestion
* Running with a shared ring and upgrading from any version less than v1.3.0 (e.g. v1.2.0) -> v1.4.0: **ACTION REQUIRED**
```

File: docs/operations/upgrade.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+There are two options for upgrade if you are not on version 1.3.0 and are using a shared ring
```
Comment (rfratto) [inline docs/operations/upgrade.md:30]:
```suggestion
There are two options for upgrade if you are not on version 1.3.0 and are using a shared ring:
```

File: docs/operations/upgrade.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+* Upgrade first to v1.3.0 BEFORE upgrading to v1.4.0
```
Comment (rfratto) [inline docs/operations/upgrade.md:32]:
```suggestion
* Upgrade first to v1.3.0 **BEFORE** upgrading to v1.4.0
```

File: docs/operations/upgrade.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+1. Add the following configuration to your ingesters command: `-ingester.normalise-tokens=true`
+1. Restart your ingesters with this config
+1. Proceed with upgrading to v1.4.0
+
+**Note:** It's also possible to enable this flag via config file, see the [docs](https://github.com/grafana/loki/tree/v1.3.0/docs/configuration#lifecycler_config)
+
+#### What will go wrong
+
+If you attempt to add a v1.4.0 ingester to a ring created by Loki v1.2.0 or older which does not have the config `-ingester.normalise-tokens=true`, the v1.4.0 ingester will remove all the entries in the ring for all the other ingesters as it cannot "see" them.
```
Comment (rfratto) [inline docs/operations/upgrade.md:46]: Maybe we should link to the config reference from here?

File: docs/operations/upgrade.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+OR 
+
+**Note:** If you are running a single binary you only need to add this flag to your single binary command.
+
+1. Add the following configuration to your ingesters command: `-ingester.normalise-tokens=true`
+1. Restart your ingesters with this config
+1. Proceed with upgrading to v1.4.0
+
+**Note:** It's also possible to enable this flag via config file, see the [docs](https://github.com/grafana/loki/tree/v1.3.0/docs/configuration#lifecycler_config)
```
Comment (cyriltovena) [inline docs/operations/upgrade.md:42]: For helm you can use add this to you value file if you're using loki

```
extraArgs:
  ingester.normalise-tokens: true
```

or if you're using loki-stack:
```
loki:
  extraArgs:
    ingester.normalise-tokens: true

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,13 +1,244 @@
...
+
+### Important Notes
+
+**Really, this is important**
+
+Before we get into new features, version 1.4.0 brings with it the first (that we are aware of) upgrade dependency.
+
+We have created a dedicated page for upgrading Loki in the [operations section of the docs](https://github.com/grafana/loki/tree/v1.4.0/docs/operations/upgrade.md)
+
+Also you should be aware we are now pruning old `master-xxxxx` docker images from docker hub, currently anything older than 90 days is removed.  **We will never remove released versions of Loki**
```
Comment (owen-d) [inline CHANGELOG.md:13]: I dislike that we do this, but we've been having some trouble here so I understand. Hopefully our problem here disappears in the future.

**Pull request merged**